### DISCUSSION TITLE
Q: Execution Log extensibility

### DISCUSSION THREAD

**Discussion by MiroslavMikus** [elsa-workflows/elsa-core#881] [Q&A]
Hi, I have found this framework yesterday. Just amazing and many thanks to the creator and all contributors. :pray:

I have a question about the Execution Log 
![image](https://user-images.githubusercontent.com/29312172/114527216-fe6b7b80-9c47-11eb-9861-9573d0dfed93.png)

1. Is there a way to set a custom message. Something like `return Outcome("ok", "here is my message");` ?
3. Could I set more than one message for an Action?
3. The last question is out of the scope, could I see the actual action execution `in` and `out` values/variables (maybe workflow variables as well)?

**Comment by sfmskywalker** [comment]
Hi, and welcome 😄 

1. Yes, custom activities and custom code/handlers can write custom events to the execution log via `IWorkflowExecutionLog`.
2. Yes, you can write as many messages as you want for a given activity.
3. Not at the moment, but this is on my list of things to do.

**Reply by MiroslavMikus** [reply]
As far as I can tell the `IWorkflowExecutionLog` is ELSA 2 feature. Looking foreward to the second version 😍

**Reply by sfmskywalker** [reply]
You're right 😄
